cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/orion/OrionBackendImpl.java

```diff
@@ -93,25 +88,45 @@ public JsonResponse deleteSubscriptionV1(String subscriptionId, String token)
         JsonResponse response = doRequest("POST", relativeURL, true, headers, subscriptionEnt);
         
         return response;   
-    }
+    } // deleteSubscriptionV1
+    
+    @Override
+    public JsonResponse getSubscriptionsV2byId(String token, 
+            String subscriptionId) throws Exception {
+        
+        // create the http header
+        ArrayList<Header> headers = new ArrayList<Header>();
+        headers = setHeaders(headers, token);
+        
+        String relativeURL = "/v2/subscriptions/" + subscriptionId;
+        JsonResponse response = doRequest("GET", relativeURL, true, headers, null);
+        
+        return response;
+    } // getSubscriptionsV2byId
     
     @Override
     public JsonResponse deleteSubscriptionV2(String subscriptionId, String token) throws Exception {
         
         // create the http header
         ArrayList<Header> headers = new ArrayList<Header>();
+        headers = setHeaders(headers, token);
+        
+        String relativeURL = "/v2/subscriptions/" + subscriptionId;
+        JsonResponse response = doRequest("DELETE", relativeURL, true, headers, null);
+        
+        return response;
+        
+    } // deleteSubscriptionV2
+    
+    private ArrayList<Header> setHeaders (ArrayList<Header> headers, String token) {
```

Comment:
As mentioned above, this could be renamed as `getHeaders(token)`:

```
private ArrayList<Header> getHeaders (String token) {
    ArrayList<Header> headers = new ArrayList<Header>();
    headers.add(new BasicHeader("Content-type", "application/json"));
    headers.add(new BasicHeader("Accept", "application/json"));
         
    if (token != null) {
        headers.add(new BasicHeader("X-Auth-token", token));
    } // if

    return headers;
} // getHeaders
```